libftl/ingest.c

```diff
@@ -92,6 +92,111 @@ OS_THREAD_ROUTINE _ingest_get_rtt(void *data) {
     return 0;
 }
 
+ftl_status_t ftl_find_closest_available_ingest(const char* ingestHosts[], int ingestsCount, char* bestIngestHostComputed)
+{
+    ftl_ingest_t* ingestElements;

```

Comment:
You should initialize these to null, otherwise your free() will be bad news bears.

Reply:
Ah. Right. this happened on XSAPI months ago, which influenced the Android app and I am doing the same. Will fix it. Thanks!